# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: editor/libeditor/EditorBase.h
# Commit: bdbed5920aeb
# Full Hash: bdbed5920aeb2adc75f03ac0f6c427b3ac456395
# Author: Jan-Niklas Jaeschke <jjaschke@mozilla.com>
# Date: 2022-10-26 03:47:46
# Regressor Bug: 1783641
# File Overlap Count: 1
# Description:
#   Bug 1783641: Deleting a selected word deletes surrounding whitespace on MacOS. r=masayuki
#   
#   Native behaviour on MacOS dictates one whitespace being removed after double-clicking a word and pressing delete.
#   This behaviour is achieved by saving the information that the selection is created by doubleclick to the `nsFrameSelection`
#   and using it in the `DeleteRangeTransaction`, where the range is extended by one whitespace character before or after the range.
# ==============================================================================

diff -r a32dbcfbd95f -r bdbed5920aeb editor/libeditor/EditorBase.h
--- a/editor/libeditor/EditorBase.h	Tue Oct 25 07:00:26 2022 +0000
+++ b/editor/libeditor/EditorBase.h	Tue Oct 25 07:00:26 2022 +0000
@@ -957,6 +957,14 @@
                              nsIPrincipal* aPrincipal = nullptr);
     ~AutoEditActionDataSetter();
 
+    void SetSelectionCreatedByDoubleclick(bool aSelectionCreatedByDoubleclick) {
+      mSelectionCreatedByDoubleclick = aSelectionCreatedByDoubleclick;
+    }
+
+    [[nodiscard]] bool SelectionCreatedByDoubleclick() const {
+      return mSelectionCreatedByDoubleclick;
+    }
+
     void UpdateEditAction(EditAction aEditAction) {
       MOZ_ASSERT(!mHasTriedToDispatchBeforeInputEvent,
                  "It's too late to update EditAction since this may have "
@@ -1353,6 +1361,10 @@
     EditorBase& mEditorBase;
     RefPtr<Selection> mSelection;
     nsTArray<OwningNonNull<Selection>> mRetiredSelections;
+
+    // True if the selection was created by doubleclicking a word.
+    bool mSelectionCreatedByDoubleclick{false};
+
     nsCOMPtr<nsIPrincipal> mPrincipal;
     // EditAction may be nested, for example, a command may be executed
     // from mutation event listener which is run while editor changes